test/mocks/genMocks.ts

```diff
@@ -0,0 +1,155 @@
+import "mocha";
+import {
+    BigNumber,
+    ContractTransaction,
+    ethers,
+    Signer,
+} from "ethers";
+import { Block, Provider } from "@ethersproject/providers";
+import { GenSale } from "../../src/contracts";
+
+export const feb23TsSeconds = 1677173138;
+export const feb24TsSeconds = 1677264314;
+
+export type GenMocks = {
+    signer: Partial<Signer>;
+    genSale: Partial<GenSale>;
+    currentSaleConfiguration: [
```

Comment:
We should wrap this type into a single interface instead of repeating it here and below

```
// types.ts
  
export interface SaleConfig {
  sellerWallet: string;
  parentDomainId: BigNumber;
  salePrice: BigNumber;
  amountForSale: BigNumber;
  mintlistMerkleRoot: string;
  startingMetadataIndex: BigNumber;
  folderGroupID: BigNumber;
  transactionLimit: BigNumber;
}
```

Or if it somehow requires the union type to be used specifically you could do

```
// types.ts
  
export type SaleConfig = [
  string,
  BigNumber,
  BigNumber,
  BigNumber,
  string,
  BigNumber,
  BigNumber,
  BigNumber
] & {
  sellerWallet: string;
  parentDomainId: BigNumber;
  salePrice: BigNumber;
  amountForSale: BigNumber;
  mintlistMerkleRoot: string;
  startingMetadataIndex: BigNumber;
  folderGroupID: BigNumber;
  transactionLimit: BigNumber;
}
```

Reply:
The union type is required. This type is only used for this function. I think an in-line type is OK here.

Reply:
It is used twice, on lines 17 and 52. It would make for nicer code if we could hide this part and have the function return some type definition for the interface I think

Reply:
Alright, I split the saleconfig off into its own interface